File: typescript-ai-agent/mcp-server-shop.ts
import { FastMCP } from "fastmcp";
import { z } from "zod";
import { getProductList, postBasket } from "./request.js";

const mcp = new FastMCP({
  name: "S&N Webshop",
  version: "1.0.0",
  instructions:
    "Provides tools for interacting with Webshop, like showing products, adding items to the cart and more.",
});

mcp.addTool({
  name: "requestProductList",
  description: "Return product list from S&N Shop.",
  parameters: z.object({}),
  execute: async () => {
    const result = await getProductList();
    return `products: ${JSON.stringify(result)}`;
  },
});

mcp.addTool({
  name: "addToBasket",
  description: "Add a product to the shopping basket.",
  parameters: z.object({
    productId: z.string().describe('Product ID, e.g. "GAD-001"'),
    quantity: z.number().int().positive().describe("Number of items to add"),
  }),
  execute: async ({ productId, quantity }) => {
    const result = await postBasket(productId, quantity);
    return `basket: ${JSON.stringify(result)}`;
  },
});

mcp.start({
  transportType: "httpStream",
  httpStream: { port: 3010 },
});

File: python-ai-agent/mcp-server-shop.py
# Fast MCP: https://gofastmcp.com/servers/server
# MCP Prtokoll io: https://modelcontextprotocol.io/docs/develop/build-server

from fastmcp import FastMCP

from request import get_product_list, get_product, get_basket, post_basket

mcp = FastMCP(
    "S&N Webshop",
    instructions="Provides tools for interacting with Webshop, like shwoing products, adding items to the cart and more.",
)

@mcp.tool
async def request_product_list() -> str:
    """Return product list from S&N Shop."""
    result = await get_product_list()
    return f'products: {result}'

def main():
    # Initialize and run the server
    mcp.run(transport="http", host="127.0.0.1", port=9000)
    
if __name__ == "__main__":
    main()

